x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -6,157 +6,143 @@
-    getMinutesPassedSinceMoment(installationPackageInfo?.install_started_at || 0)
+    calculateDiffFromNowInMinutes(installation?.install_started_at || MIN_DATE)
   );
 
   if (status === 'not-installed')
     return {
       status,
-      latestPackageVersion,
+      latestPackageVersion: latestCspPackageVersion,
       healthyAgents,
-      installedIntegrations: installedIntegrationsTotal,
+      installedPackagePolicies: installedPackagePoliciesTotal,
     };
 
   return {
     status,
-    latestPackageVersion,
+    latestPackageVersion: latestCspPackageVersion,
     healthyAgents,
-    installedIntegrations: installedIntegrationsTotal,
-    installedPackageVersion: installationPackageInfo?.install_version,
+    installedPackagePolicies: installedPackagePoliciesTotal,
+    installedPackageVersion: installation?.install_version,
   };
 };
 
-export const defineGetCspSetupStatusRoute = (router: CspRouter): void =>
-  router.get(
-    {
-      path: STATUS_ROUTE_PATH,
-      validate: false,
-      options: {
-        tags: ['access:cloud-security-posture-read'],
-      },
-    },
-    async (context, _, response) => {
-      const cspContext = await context.csp;
-      try {
-        const cspSetupStatus = await getCspSetupStatus(
-          cspContext.logger,
-          cspContext.esClient.asCurrentUser,
-          cspContext.soClient,
-          cspContext.packageService,
-          cspContext.packagePolicyService,
-          cspContext.agentPolicyService,
-          cspContext.agentService
-        );
-
-        const body: CspSetupStatus = cspSetupStatus;
-
-        return response.ok({
-          body,
-        });
-      } catch (err) {
-        const error = transformError(err);
-        cspContext.logger.error(`Error while fetching status: ${err}`);
+const getHealthyAgents = async (
+  soClient: SavedObjectsClientContract,
+  installedCspPackagePolicies: PackagePolicy[],
+  agentPolicyService: AgentPolicyServiceInterface,
+  agentService: AgentService
+): Promise<number> => {
+  // Get agent policies of package policies (from installed package policies)
+  const agentPolicies = await getCspAgentPolicies(
+    soClient,
+    installedCspPackagePolicies,
+    agentPolicyService
+  );
 
-        return response.customError({
-          body: { message: error.message },
-          statusCode: error.statusCode,
-        });
-      }
-    }
+  // Get agents statuses of the following agent policies
+  const agentStatusesByAgentPolicyId = await getAgentStatusesByAgentPolicies(
+    agentService,
+    agentPolicies
   );
+
+  // TODO: should be fixed - currently returns all agents instead of healthy agents only
+  return Object.values(agentStatusesByAgentPolicyId).reduce((sum, status) => sum + status.total, 0);
+};
+
+const calculateCspStatusCode = (
+  hasFindings: boolean,
+  installedCspPackagePolicies: number,
+  healthyAgents: number,
+  timeSinceInstallationInMinutes: number
+): CspStatusCode => {
+  if (hasFindings) return 'indexed';
```

Comment:
this is can return `indexed` status even if there are docs in `findings` but not in `latest-findings` and will result in empty and unexpected UI.